pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+
+	}
+	return clusterID, nil
+}
+
+// abstracted functionality in order to do the assume-role chaining in order to assume the support role
+// credentials in a given customer cluster. This function will get the accountclaim, assume the jump role,
+// and then from that assumed role make another assume_role call into the customer's account
+func (c *CloudClient) AssumeSupportRole() (*sts.Credentials, error) {
+
+	clusterID, err := extractClusterID(c.clusterDeployment)
+	if err != nil {
+		glog.Errorf("Could not extract ClusterID from ClusterDeployment: %s", err.Error())
+		return nil, err
+	}
+
+	glog.V(0).Infof("Getting AccountClaim for %s", clusterID)
+	ac, err := c.ocmClient.GetAWSAccountClaim(clusterID)
+	glog.V(0).Infof("Received %+v", ac)
+	if err != nil || ac == nil {
+		return nil, &ccerrors.UpstreamError{Err: fmt.Errorf("could not obtain an accountclaim from OCM")}
+	}
+
+	glog.V(0).Info("Validating AccountClaim")
+	err = c.validateAccountClaim(ac)
+	if err != nil {
+		return nil, &ccerrors.ValidationError{Err: err}
+	}
+	glog.V(0).Info("AccountClaim Validated")
+
+	jumpRole := jumpRole
+	glog.V(0).Infof("Obtained Jump Role ARN %s from configmap", jumpRole)
+
+	sessionName := fmt.Sprintf("cad-check")
+	glog.Info("Building aws session for ", sessionName)
+
+	glog.V(0).Info("Assuming role into the jump role ", jumpRole)
+	// We don't need to limit the scope of the jump role, as the only permission the Jump role should
+	// have is AssumeRole, so we pass in `nil` as the role here.
+	jrCreds, err := c.getAssumedRoleCredentials(c.awsClient, jumpRole, sessionName)
+	if err != nil {
+		return nil, &ccerrors.UpstreamError{Err: err}
+	}
+	if jrCreds == nil {
+		return nil, &ccerrors.UpstreamError{
+			Err: fmt.Errorf("Jump Role Assume returned empty credentials"),
+		}
+	}
+	jumpRoleClient, err := c.getAwsClientBuilder().New(*jrCreds.AccessKeyId, *jrCreds.SecretAccessKey, *jrCreds.SessionToken, c.region)
+	if err != nil {
+		return nil, &ccerrors.UpstreamError{
+			Err: fmt.Errorf("Could not build new jump role client with assumed creds: %s", err.Error()),
+		}
+	}
+
+	glog.V(0).Info("Assuming Support Role ", ac.Spec.SupportRoleARN)
+	cuCreds, err := c.getAssumedRoleCredentials(jumpRoleClient, ac.Spec.SupportRoleARN, sessionName)
+	if err != nil {
+		return nil, &ccerrors.RequestError{
+			Err: fmt.Errorf("Could not assume support role in customer's account: %s", err.Error()),
+		}
+	}
+	return cuCreds, nil
+}
+
+func (c *CloudClient) validateAccountClaim(ac *awsv1alpha1.AccountClaim) error {
+	supportRoleArn := ac.Spec.SupportRoleARN
+	if supportRoleArn == "" {
+		// if the supportRoleARN is not set, then we won't know which role inside of the customer
+		// AWS account to assume into. This is defined by the customer for STS clusters, and defined
+		// by the aws-account-operator on CCS and OSD accounts
+		return errors.Wrap(fmt.Errorf("AccountClaim is invalid"), "supportRoleARN is not present in the AccountClaim")
```

Comment:
Change this to fmt.Errorf

This way we can remove the `errors` package and not alias `ccerrors`